db/columns.py

```diff
@@ -240,15 +269,52 @@ def change_column_nullable(table_oid, column_index, nullable, engine):
     return tables.reflect_table_from_oid(table_oid, engine).columns[column_index]
 
 
-def drop_column(
-        engine,
-        table_oid,
-        column_index,
-):
+def drop_column(table_oid, column_index, engine):
     column_index = int(column_index)
     table = tables.reflect_table_from_oid(table_oid, engine)
     column = table.columns[column_index]
     with engine.begin() as conn:
         ctx = MigrationContext.configure(conn)
         op = Operations(ctx)
         op.drop_column(table.name, column.name, schema=table.schema)
+
+
+def get_column_default(table_oid, column_index, engine):
+    table = tables.reflect_table_from_oid(table_oid, engine)
+    column = table.columns[column_index]
+    return _decode_server_default(column.server_default, engine)
+
+
+def _decode_server_default(server_default, engine):
+    if server_default is not None:
+        cast_sql_text = server_default.arg.text
+        with engine.begin() as conn:
+            # Defaults are returned as text with SQL casts appended
+            # Ex: "'test default string'::character varying" or "'2020-01-01'::date"
+            # Here, we execute the cast to get the proper python value
+            return conn.execute(select(text(cast_sql_text))).first()[0]
```

Comment:
This is creating a problem for any column with a serial default, since it actually pulls one from the underlying sequence.  To see this, reload the column list for some table, and watch the `mathesar_id` increment.  The values are burnt when they're shown, BTW, so inserts won't actually use the displayed value, but rather the one after that.

I suggest completely avoiding dealing with dynamically-generated defaults for the moment (unless they're something we're dealing with ourselves).  Alternatively, we could try to come up with some supported ones (e.g., basic integer sequences, pseudo-random UUIDs, `NOW()` for time and date columns), and create separate stories for each of those.  I think they'll need to be set up with some bespoke code.

Regardless, the behavior with sequences is at best confusing (since any inserted row will actually not get the index that's shown in the API, but rather the next value after that), and should be fixed.  I also think that regardless of the confusing behavior, we should avoid incrementing sequences that Mathesar may not have created (i.e., the user could import a table that has its own sequences), since this burns values from the sequence, which could create issues under some (admittedly rare) circumstances for users.

Reply:
I'm in support of avoiding dealing with them for now. We can figure out if/when we want to support dynamically-generated defaults later.

Reply:
I've updated to check for the textual SQL starts with `nextval(` and ends with `)`. This seems somewhat brittle, but given all strings should be wrapped by `'`s I think we should be safe. We could instead check the sqlalchemy column is a primary key, is an integer and has `autoincrement` set to true (the conditions for sqlalchemy creating a serial column). But, this would fail on serial columns manually set on the server, so I figured the current solution was best.

Reply:
I think the most robust ways would be to either:

- check the `attmissingval` field of the `pg_attribute` entry for the column for a non-null value (this should only be non-null if there's a "non-volatile `DEFAULT`, and has the advantage of actually serving up the value), or
- check the `atthasmissing` field of the `pg_attrdef` entry for the column for the prefix `{CONST`, which is part of the expression that actually "generates" the column, or
- check the boolean field `atthasmissing` of the `pg_attribute` entry for the column (I _think_ this should only be true for non-volatile `DEFAULT` values; please double-check that in the docs), and then get the actual default separately.  

Unfortunately, the columns involved in the first two options are of types not supported by SQLAlchemy (`anyarray` and `pg_node_tree` respectively), so you'd need to handle the return value as text.  I don't think that should be too bad, though.  My favorite option is the first bullet point above, since you can use a single query to both verify whether there's a (non-volatile) `DEFAULT`, and also get the value itself with no further checking.  The only issue will be getting the right type for the value, but that should be in your hands (check `Column.type.python_type`) when you're looking up the value anyway.